set-gtk-dark-theme.py

```diff
@@ -0,0 +1,162 @@
+                        and win_name != "Discord Updater"
+                        and win_class
+                        and not set(win_class).isdisjoint(window_classes)
+                    ):
+                        return set_theme_variant_by_window_id(win_id, variant)
+            except Xlib.error.BadWindow:
+                pass
+    return False
+
+
+def set_theme_variant_by_window_id(window_id, variant):
+    # Pretty self explanatory. Use subprocess to call
+    # xprop and set the variant.
+    output = subprocess.run(
+        [
+            "xprop",
+            "-id",
+            str(window_id),
+            "-f",
+            "_GTK_THEME_VARIANT",
+            "8u",
+            "-set",
+            "_GTK_THEME_VARIANT",
+            variant,
+        ]
+    )
+    return output.returncode == 0
+
+
+def set_theme_variant_by_window_title(title, variant):
+    # Pretty self explanatory. Use subprocess to call
+    # xprop and set the variant.
+    output = subprocess.run(
+        [
+            "xprop",
+            "-name",
+            title,
+            "-f",
+            "_GTK_THEME_VARIANT",
+            "8u",
+            "-set",
+            "_GTK_THEME_VARIANT",
+            variant,
+        ]
+    )
+    return output.returncode == 0
+
+
+def fallback(window_titles, variant):
+    # In the event that the
+    # window we were looking for
+    # doesn't show up in a timely manner
+    # fallback to blindly trying to set
+    # the variant.
+    for title in window_titles:
+        set_theme_variant_by_window_title(title, variant)
+
+
+def main():
+    start = time.time()
+    # Listen for X Property Change events.
+    root.change_attributes(event_mask=Xlib.X.PropertyChangeMask)
+    window_titles = window_classes = ("Discord", "discord")
+    variant = "dark"
+    # In the unlikely event that our window shows up before
+    # the script starts don't start a while loop.
+    if not set_theme_variant(window_titles, window_classes, variant):
+        # Basically give our window 5 secs to show up.
+        # If it doesn't just fallback to the old way of blindly
+        # trying to set the theme variant
+        while True:
+            if time.time() - start <= 5:
+                # disp.next_event() blocks if no events are
+                # queued. In combination with while True
+                # it creates a very simple event loop.
+                disp.next_event()
+                # We only try to set the variant after
+                # there has been a Property Change event.
+                if set_theme_variant(window_titles, window_classes, variant):
+                    break
+            else:
+                fallback(window_titles, variant)
+                break
+
+
+if __name__ == "__main__":
+    xdg_config_home = os.environ.get("XDG_CONFIG_HOME") or os.path.join(
+        os.path.expanduser("~"), ".config"
+    )
+    with open(
```

Comment:
All of this should be robust against exceptions. Opening the file, parsing the json, etc.

Reply:
Thanks for pointing that out. I've added a try block to the script that should cover the most common exceptions.